services/alchemy_vision/v1.js

```diff
@@ -14,64 +14,171 @@
 var FEATURE_RESPONSES = {
   imageFaces: 'imageFaces',
   imageLink: "image",
-  imageKeywords: "imageKeywords"
+  imageKeywords: "imageKeywords",
+  imageText: "sceneTextLines"
 };
 
+
 module.exports = function (RED) {
-  var cfenv = require('cfenv'),
-    AlchemyAPI = require('alchemy-api');
+  var cfenv = require('cfenv');
+  var watson = require('watson-developer-cloud');
+
+  var imageType = require('image-type');
+  var url = require('url');
+  var temp = require('temp');
+  var fileType = require('file-type');
+  var fs = require('fs');
+
+  // temp is being used for file streaming to allow the file to arrive so it can be processed. 
+  temp.track();
+
+  // Require the Cloud Foundry Module to pull credentials from bound service 
+  // If they are found then the api key is stored in the variable s_apikey. 
+  //
+  // This separation between s_apikey and apikey is to allow 
+  // the end user to modify the key  redentials when the service is not bound.
+  // Otherwise, once set apikey is never reset, resulting in a frustrated
+  // user who, when he errenously enters bad credentials, can't figure out why
+  // the edited ones are not being taken.
 
-  var services = cfenv.getAppEnv().services,
-    service;
+  // Taking this line out as codacy was complaining about it. 
+  // var services = cfenv.getAppEnv().services,
 
-  var apikey;
+  var apikey, s_apikey;
 
   var service = cfenv.getAppEnv().getServiceCreds(/alchemy/i);
 
   if (service) {
-    apikey = service.apikey;
+    s_apikey = service.apikey;
   }
 
   RED.httpAdmin.get('/alchemy-image-analysis/vcap', function (req, res) {
     res.json(service ? {bound_service: true} : null);
   });
 
+  // Utility functions that check for image buffers, urls and stream data in
+ 
+  function imageCheck(data) {
+    return data instanceof Buffer && imageType(data) !== null;
+  };
+
+  function urlCheck(str) {
+    var parsed = url.parse(str)
+    return (!!parsed.hostname && !!parsed.protocol && str.indexOf(' ') < 0);
+  };
+
+  function stream_buffer(file, contents, cb) {
+    fs.writeFile(file, contents, function (err) {
+      if (err) throw err;
+      cb();
+    });
+  };
+
+  // Utility function that performs the alchemy vision call. 
+  // the cleanup removes the temp storage, and I am not sure whether 
+  // it should be called here or after alchemy returns and passed
+  // control back to cbdone.
+
+  function performAction(params, feature, cbdone, cbcleanup) {
+    var alchemy_vision = watson.alchemy_vision( { api_key: apikey } );
+
+    if (feature == 'imageFaces')
+    {
+      alchemy_vision.recognizeFaces(params, cbdone);
+    } else if (feature == 'imageLink') {
+      alchemy_vision.getImageLinks(params, cbdone);
+    } else if (feature == 'imageKeywords') {
+      alchemy_vision.getImageKeywords(params, cbdone);
+    } else if (feature == 'imageText') {
+      alchemy_vision.getImageSceneText(params, cbdone);
+    }    
+
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Expected '{' and instead saw 'cbcleanup'.](https://www.codacy.com/app/rezgui-y/node-red-node-watson/file/2569415486/issues/source?bid=3225115&fileBranchId=3297455#l100)